### PR TITLE
Support an overrides.proto file to load system specific PID data.

### LINKED ISSUE #732: Allow for overriding PID definitions.
Use case:

A manufacturer has a custom PID that is normally GET only. However in the factory, they use the OLA tools to send SET commands as part of programming the unit. We don't want to publish the SET command on the rdm.openlighting.org website since it shouldn't be used outside the factory.

The proposal is to allow an *overrides.proto* file to override the PID definitions usually found in pids.proto & manufacturer_pids.proto . The manufacturer can then write their own overrides.proto with the SET command and use the tools as required. 

I think this is a pretty small change to the PidLoader.

### PR DESCRIPTION
**Pull request by nomis52**
Closes #732

### CODE REVIEW

File: common/rdm/PidStoreLoader.cpp [hunk 1]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
                           manufacturer_map,
                           store_pb.version());
 }
 
+/*
+ * @brief Load the data from the PidStore proto into the ManufacturerMap.
+ * @param[out] pid_data the ManufacturerMap to populate.
+ * @param proto the Protobuf data.
+ * @param validate Enables strict validation mode.
+ * If a collision occurs, the data in the map is not replaced.
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: @note or leave a blank line

File: common/rdm/PidStoreLoader.cpp [hunk 2]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
+    if (!GetPidList(iter->second, manufacturer, validate, false)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+/*
+ * @brief Populate a PidMap from a protobuf object that has a set of repeated
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: You've moved all the other Doxygen docs out, why not this?

File: common/rdm/PidStoreLoader.cpp [hunk 3]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
     OLA_DEBUG << "Loading " << pid.name();
     if (validate) {
-      set<uint16_t>::const_iterator value_iter = seen_values.find(pid.value());
-      if (value_iter != seen_values.end()) {
-        OLA_WARN << "Pid " << pid.value() << " exists multiple times in the "
-            "pid file";
-        ok = false;
-        break;
+      if (STLContains(seen_pids, pid.value())) {
+        OLA_WARN << "Pid " << pid.value()
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: SPaG PID

File: common/rdm/PidStoreLoader.cpp [hunk 4]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
+                 << " exists multiple times in the pid file";
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: Again

File: common/rdm/PidStoreLoader.cpp [hunk 5]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
-      set<string>::const_iterator name_iter = seen_names.find(pid.name());
-      if (name_iter != seen_names.end()) {
-        OLA_WARN << "Pid " << pid.name() << " exists multiple times in the "
-            "pid file";
-        ok = false;
-        break;
+      seen_pids.insert(pid.value());
+
+      if (STLContains(seen_names, pid.name())) {
+        OLA_WARN << "Pid " << pid.name()
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: PID

File: common/rdm/PidStoreLoader.cpp [hunk 6]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
+                 << " exists multiple times in the pid file";
+        return false;
       }
       seen_names.insert(pid.name());
 
       if (limit_pid_values && pid.value() > 0x8000 && pid.value() < 0xffe0) {
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: Are these not defined already, if not we should make them constants. Or add a pid.isvalid()

File: common/rdm/PidStoreLoader.cpp [hunk 7]
```diff
@@ -127,126 +121,168 @@ const RootPidStore *PidStoreLoader::LoadFromStream(std::istream *data,
...
     }
-    return false;
+    iter->second = descriptor;
   }
   return true;
 }
 
-
 /**
  * Build a PidDescriptor from a Pid protobuf object
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: @brief

File: common/rdm/PidStoreLoader.h [hunk 8]
```diff
@@ -39,52 +40,88 @@ namespace rdm {
...
  public:
-    PidStoreLoader() {}
-    ~PidStoreLoader() {}
+  PidStoreLoader() {}
 
-    // Load information into this store
-    const RootPidStore *LoadFromFile(const std::string &file,
-                                     bool validate = true);
+  /**
+   * @brief Load Pid information from a file.
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.h]: PID

File: common/rdm/PidStoreLoader.h [hunk 9]
```diff
@@ -39,52 +40,88 @@ namespace rdm {
...
+   * @returns A pointer to a new RootPidStore or NULL if loading failed.
+   */
+  const RootPidStore *LoadFromFile(const std::string &file,
+                                   bool validate = true);
 
-    // Load information into this store
-    const RootPidStore *LoadFromDirectory(const std::string &directory,
-                                          bool validate = true);
+  /**
+   * @brief Load Pid information from a directory, including overrides.
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.h]: Again (or is this the class name?

**Comment by peternewman** [comment]
Add overrides.proto to .gitignore.

**Comment by peternewman** [comment]
LGTM generally

File: common/rdm/PidStoreLoader.cpp (hunk 1 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Done

File: common/rdm/PidStoreLoader.cpp (hunk 2 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: This is an internal method. I removed the /**

File: common/rdm/PidStoreLoader.cpp (hunk 3 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Done

File: common/rdm/PidStoreLoader.cpp (hunk 4 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Done

File: common/rdm/PidStoreLoader.cpp (hunk 5 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Done

File: common/rdm/PidStoreLoader.cpp (hunk 2 above)
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: Ah yes.;

File: common/rdm/PidStoreLoader.cpp (hunk 6 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Nothing else differentiates between ESTA & Manufacturer PIDs. I made them constants.

File: common/rdm/PidStoreLoader.cpp (hunk 7 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.cpp]: Removed /**

File: common/rdm/PidStoreLoader.h (hunk 8 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.h]: Done

File: common/rdm/PidStoreLoader.h (hunk 9 above)
Comment (nomis52) [inline common/rdm/PidStoreLoader.h]: Done

File: common/rdm/PidStoreLoader.cpp [hunk 10]
```diff
@@ -46,13 +48,11 @@ using std::set;
 using std::string;
 using std::vector;
 
+const char PidStoreLoader::OVERRIDE_FILE_NAME[] = "overrides.proto";
+const uint16_t PidStoreLoader::ESTA_MANUFACTURER_ID = 0;
+const uint16_t PidStoreLoader::MANUFACTURER_PID_MIN = 0x8000;
```
Comment (peternewman) [inline common/rdm/PidStoreLoader.cpp]: I meant in RDMConstant.h

**Pull request merged**